examples/ingestron/main.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"encoding/json"
+	"os/signal"
+	"syscall"
+	"time"
+        r "gopkg.in/dancannon/gorethink.v2"
+	"github.com/go-steem/rpc"
+	"github.com/go-steem/rpc/transports/websocket"
+)
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatalln("Error:", err)
+	}
+}
+
+// Set the settings for the DB
+func run() (err error) {
+    Rsession, err := r.Connect(r.ConnectOpts{
+    Addresses: []string{"192.168.194.98:28015","192.168.194.91:28015","192.168.194.145:28015","192.168.194.209:28015"},
+    })
+    if err != nil {
+        log.Fatalln(err.Error())
+    }
+
+// Create a table in the DB
+var rethinkdbname string = "steemit"
+_, err = r.DBCreate(rethinkdbname).RunWrite(Rsession)
+Rsession.Use(rethinkdbname)
+	if err != nil {
+		fmt.Println("rethindb DB already made")
+}
+	_, err = r.DB(rethinkdbname).TableCreate("blocks").RunWrite(Rsession)
+
+	if err != nil {
+		fmt.Println("Probably already made a table for blocks")
+
+	}
+
+
+	// Process flags.
+	flagAddress := flag.String("rpc_endpoint", "ws://138.201.198.167:8090", "steemd RPC endpoint address")
+	flagReconnect := flag.Bool("reconnect", false, "enable auto-reconnect mode")
+	flag.Parse()
+
+	var (
+		url       = *flagAddress
+		reconnect = *flagReconnect
+	)
+
+	// Start catching signals.
+	var interrupted bool
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM)
+
+	// Drop the error in case it is a request being interrupted.
+	defer func() {
+		if err == websocket.ErrClosing && interrupted {
+			err = nil
+		}
+	}()
+	// This allows you to tell the app which block to start on.
+	// TODO: Make all of the vars into a config file and package the binaries
+	Startblock := 3100000
+	U := uint32(Startblock)
+	// Start the connection monitor.
+	monitorChan := make(chan interface{}, 1)
+	if reconnect {
+		go func() {
+			for {
+				event, ok := <-monitorChan
+				if ok {
+					log.Println(event)
+				}
+			}
+		}()
+	}
+
+	// Instantiate the WebSocket transport.
+	log.Printf("---> Dial(\"%v\")\n", url)
+	t, err := websocket.NewTransport(url,
+		websocket.SetAutoReconnectEnabled(reconnect),
+		websocket.SetAutoReconnectMaxDelay(30*time.Second),
+		websocket.SetMonitor(monitorChan))
```

Comment:
@faddat I don't think that you need to use auto-reconnect and monitoring, it was just a demo in the example you based this code on. IMO it makes more sense to just drop this and also delete the extra code that is associated with this functionality. Examples should do as little as possible to be examples really...